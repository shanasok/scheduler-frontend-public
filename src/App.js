import React from 'react';
import { connect } from "react-redux";
import {Helmet} from "react-helmet";

import { bindActionCreators } from "redux";
import { setCoursesJson } from './redux/actions/coursesAction';
import { getCourses } from "./utils/coursesFetcherGraphQL";
import Layout from './container/Layout.js';

import './App.css';
import {setEditSlideFalse} from "./redux/actions/editSlideAction";

class App extends React.Component {

    componentDidMount() {
        getCourses(this.props.actions.setCoursesJson, this.props.actions.setEditSlideFalse)
    }

    render() {
        return (
            <div className="App">
                <Helmet>
                    <script src="https://kit.fontawesome.com/ae667b96c5.js" crossOrigin="anonymous"></script>
                </Helmet>
                <Layout />
            </div>
        );
    }
}

function mapStateToProps(state) {
    return {
        courses: state.courses
    };
}

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            setCoursesJson,
            setEditSlideFalse,
        }, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(App)
