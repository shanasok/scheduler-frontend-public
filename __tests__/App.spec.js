import renderer from "react-test-renderer";
import {Provider as ReduxProvider} from "react-redux";
import App from "../src/App";
import React from "react";
import configureMockStore from "redux-mock-store";

const mockStore = configureMockStore()
const mockState = {}
let store = mockStore(mockState)

describe('My Test Suite', () => {
    it('My Test Case', () => {
        expect(true).toEqual(true);
    });

    it('renders correctly', () => {
        const tree = renderer
            .create( <ReduxProvider store={store}><App/></ReduxProvider>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

});