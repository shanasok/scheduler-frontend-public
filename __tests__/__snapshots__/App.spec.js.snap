// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Test Suite renders correctly 1`] = `
<div
  className="App"
>
  <div
    id="layoutDiv"
  >
    <header>
      Shana's Lessons
    </header>
    <div
      id="main"
    >
      <div
        id="mainCourseChooser"
      >
        <div
          id="mainTitle"
        >
          Click to see course content, materials and scheduling information.
        </div>
        <div
          id="mainCourses"
        />
        <div
          id="mainBottom"
        >
          Main Bottom goes here
        </div>
      </div>
    </div>
    <footer>
      My Footer here
    </footer>
  </div>
</div>
`;
